### PR TITLE
[12] [UI] As a user, I can see survey detail screen

### PR DESCRIPTION
**Pull request by manh-t**
https://github.com/manh-t/flutter-survey/issues/12

## What happened 👀

After clicking Next button at Home screen, I can navigate to Survey detail screen.

## Proof Of Work 📹

https://user-images.githubusercontent.com/60863885/118347820-3c71fd00-b570-11eb-8eeb-1064d4bd8f1c.mp4

### CODE REVIEW

File: lib/pages/surveydetail/survey_detail_page.dart [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  Widget build(BuildContext context) {
+    final survey = ModalRoute.of(context)!.settings.arguments as Survey;
+    return Scaffold(
+      body: Stack(
+        children: [
+          Container(
+            decoration: BoxDecoration(
+              image: DecorationImage(
+                image: CachedNetworkImageProvider(
+                    (survey.coverImageUrl ?? "") + "l"),
```
Comment (markgravity) [inline lib/pages/surveydetail/survey_detail_page.dart:22]: What does this `"l"` use for? 🤔

File: lib/pages/surveydetail/survey_detail_page.dart [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                  colors: [
+                    Colors.black.withOpacity(0.3),
+                    Colors.black.withOpacity(0.5),
+                  ],
+                  stops: [
+                    0.0,
+                    0.6,
+                  ]),
+            ),
+          ),
```
Comment (markgravity) [inline lib/pages/surveydetail/survey_detail_page.dart:42]: I saw you duplicate this code pretty much time to create a background with dimmed effect. 
What do you think about adding a reusable widget there? Ex: `CoverBackgroundWidget` 🤔

File: lib/pages/surveydetail/survey_detail_page.dart (hunk 1 above)
Comment (manh-t) [reply lib/pages/surveydetail/survey_detail_page.dart:22]: image
😄

Comment (markgravity) [reply lib/pages/surveydetail/survey_detail_page.dart:22]: This is the tip of the day 👍

File: lib/pages/dashboard/widget/dashboard_footer.dart [hunk 3]
```diff
@@ -77,8 +80,11 @@ class DashboardFooter extends StatelessWidget {
                             fit: BoxFit.none,
                           )),
                       onTap: () {
-                        ScaffoldMessenger.of(context)
-                            .showSnackBar(SnackBar(content: Text("Pressed")));
+                        final currentSurvey = context
+                            .read(dashboardModelProvider)
+                            .surveys[currentPage - 1];
+                        Navigator.pushNamed(context, Routes.SURVEY_DETAIL_PAGE,
+                            arguments: currentSurvey);
```
Comment (manh-t) [inline lib/pages/dashboard/widget/dashboard_footer.dart:87]: Need to fix

**Comment by manh-t** [comment]
Rebased and fix getting survey according to updated implementation in #34

File: lib/resouces/assets.dart [hunk 4]
```diff
@@ -4,4 +4,5 @@ class Assets {
   static const String bgLogin = "assets/images/bg_login.jpg";
   static const String icNimbleLogo = "assets/icons/ic_nimble_logo.svg";
   static const String icArrowRight = "assets/icons/ic_arrow_right.svg";
+  static const String icArrowLeft = "assets/icons/ic_arrow_left.svg";
```
Comment (sleepylee) [inline lib/resouces/assets.dart:7]: it's the chevron_left `Icons.chevron_left` next time you might want to use the native icon package 😁

File: lib/pages/surveydetail/survey_detail_page.dart [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import 'package:flutter_survey/pages/common/dimmed_background.dart';
+import 'package:flutter_survey/pages/login/widget/rounded_button_widget.dart';
+import 'package:flutter_survey/resouces/app_colors.dart';
+import 'package:flutter_survey/resouces/assets.dart';
+import 'package:flutter_svg/svg.dart';
+
+class SurveyDetailPage extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    final survey = ModalRoute.of(context)!.settings.arguments as Survey;
```
Comment (sleepylee) [inline lib/pages/surveydetail/survey_detail_page.dart:14]: perhaps it's better to pass only the id and fetch the survey detail here later 🤔

File: lib/pages/surveydetail/survey_detail_page.dart [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+          Positioned(
+            bottom: 40,
+            right: 30,
+            child: Container(
+              height: 56,
+              child: RoundedButtonWidget(
+                buttonText: AppLocalizations.of(context)!.surveyStartSurveyText,
+                buttonColor: Colors.white,
+                textColor: AppColors.blackRussian,
+                onPressed: () {},
```
Comment (sleepylee) [inline lib/pages/surveydetail/survey_detail_page.dart:72]: optional: better add TODO for the later quick retrieval 😅

**Review by sleepylee — APPROVED** [review]
only some minor comments

File: lib/resouces/assets.dart (hunk 4 above)
Comment (manh-t) [reply lib/resouces/assets.dart:7]: I will remember this keyword, thanks @sleepylee 👍

File: lib/pages/surveydetail/survey_detail_page.dart (hunk 5 above)
Comment (manh-t) [reply lib/pages/surveydetail/survey_detail_page.dart:14]: @sleepylee ah, I intend to fetch survey detail when pressing the `Start Survey` button 😄  sth like this: 

https://user-images.githubusercontent.com/60863885/124255484-68fbcb80-db54-11eb-893a-881081fbbe58.mp4

File: lib/pages/surveydetail/survey_detail_page.dart (hunk 6 above)
Comment (manh-t) [reply lib/pages/surveydetail/survey_detail_page.dart]: @sleepylee added, thank you 🙏

**Pull request merged**